### PR TITLE
Split build tests and run tests steps in run-ci

### PR DESCRIPTION
**Pull request by yxun**
Signed-off-by: Yuanlin <yuanlin.xu@redhat.com>

Commit Message:
Split build tests and run tests steps in CI

Additional Description:
This change is useful for reducing the resources usage when compiling and running tests.
I tested this change can avoid CPU usage spike in my OCP environment and help control the resources in the limited request level.

Risk Level: low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: maistra/run-ci.sh [hunk 1]
```diff
@@ -14,6 +14,11 @@ time bazel build \
 echo "Build succeeded. Binary generated:"
 bazel-bin/source/exe/envoy-static --version
 
+# Build tests
+time bazel build \
```
Comment (oschaaf) [inline maistra/run-ci.sh:22]: Can we add a doc comment to explain why we do split the build and test steps for future readers?

File: maistra/run-ci.sh [hunk 2]
```diff
@@ -14,6 +14,11 @@ time bazel build \
...
+  ${COMMON_FLAGS} \
+  //test/...
```
Comment (jwendell) [inline maistra/run-ci.sh:25]: Perhaps add `--build_tests_only` too?

**Review by jwendell — COMMENTED** [review]
Was this of any help at all?

File: maistra/run-ci.sh (hunk 2 above)
Comment (yxun) [reply maistra/run-ci.sh:25]: I tried this change with local disk cache. It helps avoid a CPU usage spike. 
hm, I assume all files under the //test/ are tests. I can add --build_tests_only in that line

**Review by oschaaf — APPROVED** [review]

**Pull request closed without merging**